tests/test_bayes_search.py

```diff
@@ -55,7 +55,7 @@ def run_bayes_search(
     if optimium is not None:
         best_run = (min if opt_goal == "minimize" else max)(
             [r for r in runs if r.state == run_state],
-            key=lambda run: run.metric_extremum(metric_name, opt_goal),
+            key=lambda run: run.metric_extremum(metric_name, opt_goal),  # type: ignore
```

Comment:
@dannygoldstein  as far as I can tell there is no version of mypy where this line passes for both python 3.7 and 3.10. I can get both to pass with seperate versions, but not together...